src/content/getting-started/install-an-application/_index.md

```diff
@@ -15,4 +15,238 @@ user_questions:
+  --organization=giantswarm \
+  --app-name=test01-ingress-nginx \
+  --name=ingress-nginx \
+  --target-namespace=kube-system \
+  --version=3.9.2 > ingress-nginx.yaml
+```
+
+You can push the generated App CR file to your GitOps pipeline or apply it directly to the cluster with the following command:
+
+```nohighlight
+kubectl apply -f ingress-nginx.yaml
+```
+
+After few second you should see the ingress controller running in the cluster:
+
+```nohighlight
+kubectl get app ingress-nginx
+NAME            INSTALLED VERSION   CREATED AT   LAST DEPLOYED   STATUS
+test01-ingress-nginx   3.9.2               16s          14s             deployed
+```
+
+## Step 3: Deploy hello-world app
+
+The next step is deploying the `hello-world` application using the same approach as we did with the ingress controller.
+
+First we list the versions with:
+
+```nohighlight
+kubectl get appcatalogentries -n default -l app.kubernetes.io/name=hello-world
+NAME                           CATALOG      APP NAME      VERSION   UPSTREAM VERSION   AGE
+...
+giantswarm-hello-world-2.3.2   giantswarm   hello-world   2.3.2     0.2.1              72d
+```
+
+And later we template the application to target the right organization and cluster:
+
+```nohighlight
+kubectl gs template app \
+  --catalog=giantswarm \
+  --cluster-name=test01 \
+  --organization=giantswarm \
+  --name=test01-hello-world \
+  --target-namespace=default \
+  --version=2.3.2 > hello-world.yaml
+```
+
+Lets take a look at the `App CR` to learn main fields available:
+
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+metadata:
+  name: test01-hello-world
+  namespace: org-giantswarm
+spec:
+  catalog: giantswarm
+  kubeConfig:
+    inCluster: false
+  name: hello-world
+  namespace: kube-system
+  version: 2.3.2
+```
+
+The `name` field is the name of the app in the catalog meanwhile `--app-name` designates the name of the app instance installed in the cluster. Keep in mind that the app name is subject to different length limits, depending on how the app is deployed. Using a name under 30 characters is recommended.
+
+After few seconds you will see the deployed status:
+
+```nohighlight
+kubectl get app test01-hello-world
+NAME                INSTALLED VERSION   CREATED AT   LAST DEPLOYED   STATUS
+test01-hello-world   2.3.2               11s          8s              deployed
+```
+
+Now you can access your workload cluster, check the app running, and see which ingress address has been generated:
+
+```nohighlight
+kubectl get ingress -n default
+NAME          CLASS   HOSTS                                        ADDRESS                                                                       PORTS     AGE
+hello-world   nginx   cluster.provider.k8s.giantswam.io   ab49484.cn-north-1.elb.amazonaws.com.cn   80, 443   2m29s
+```
+
+Sadly, the address isn't accessible because we've not passed the right domain on to the ingress controller.
+
+## Step 4: Configuring the right domain
```

Comment:
These sections must be merged, and the "oh sorry we described an incomplete step" part must go away. Tutorials must be concise and correct, explaining traps before they happen.

Let's remove the above app templating (including the China domain) and go on below.

Reply:
ok I will do that